### PR TITLE
Creating Settings.json spec under preview

### PR DESCRIPTION
**Pull request by thramu**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json
Comment (visingla-ms) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json:1]: As shown in the Breaking Changes report in your swagger PR, adding new paths/operations in an existing preview/stable version is a breaking change. So, you need to add new paths/operations in a new preview version.

In case, you have justification to include new paths in an existing API version, reach out to Azure Breaking Changes Reviewers <azbreakchangereview@microsoft.com> for exception and approval, and then, ARM can review the changes afterwards. Once you have the approval, please remove the "ARMChangesRequested" label from the PR to make the PR visible to the ARM reviewers.

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json
Comment (visingla-ms) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json:287]: >"description": "The scope associated with setting. This includes 'subscriptions/{subscriptionId}' for subscription scope, 'providers/Microsoft.Billing/billingAccounts/{billingAccountId}' for Billing Account scope, 'providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingProfiles/{billingProfileId}' for billing profile scope.", [](http://example.com/codeflow?start=6&length=342)

Does it mean that: all the APIs defined using {scope} in Operation Path, will exist and will be available to customers for all these 3 scopes: subscriptions, billingAccounts and billingProfiles?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json
Comment (visingla-ms) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json:195]: >OK. The request has succeeded. [](http://example.com/codeflow?start=28&length=30)

Suggestion for the descriptions:
200 means: Settings Type is deleted successfully.
204 means: Settings Type is already deleted or not found.

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json
Comment (visingla-ms) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json:247]: See the use of "x-ms-client-flatten" to avoid undesired nested level of properties. See: [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-client-flatten](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-client-flatten).

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json
Comment (visingla-ms) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json:274]: >preferContainerTags [](http://example.com/codeflow?start=9&length=19)

Suggestion: Using enum instead of boolean is always recommended, to avoid binary nature of booleans, causing breaking changes in future, when needed to extend further.

For example:
enum preferredTags {
    Container
    ...
}

or even,
enum preferContainerTags {
    True
    False
    ...
}

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json
Comment (thramu) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json:1]: This is existing preview api, we are isolating it into different file. This version was never published even it was added into the swagger.

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json
Comment (thramu) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json:287]: Thats correct

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json
Comment (thramu) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json:195]: Thanks for the valuable feedback. we need to implement this change and deploy and then make this swagger update. I will create a task for this suggestion in our backlog.

Comment (thramu) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json:195]: Thanks, Will implement as suggested

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json
Comment (thramu) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json:247]: will keep it as-is for now.

**Comment by sjanamma** [comment]
approved

**Comment by lirenhe** [comment]
@ArcturusZhang  & @msyyc , could you help to check whether the GO & Python SDK breaking changes are expected?

**Comment by thramu** [comment]
> @thramu, could you fix the LintDiff errors? thanks



> @thramu, could you fix the LintDiff errors? thanks

Can you please help me understand what is the fix for the LintDiff errors?
image

**Comment by lirenhe** [comment]
> > @thramu, could you fix the LintDiff errors? thanks
> 
> > @thramu, could you fix the LintDiff errors? thanks
> 
> Can you please help me understand what is the fix for the LintDiff errors? image

@jianyexi, could you help @thramu on how to fix this problem?   Is this a valid linter issue?

**Comment by thramu** [comment]
> > > @thramu, could you fix the LintDiff errors? thanks
> > 
> > 
> > > @thramu, could you fix the LintDiff errors? thanks
> > 
> > 
> > Can you please help me understand what is the fix for the LintDiff errors? image
> 
> @jianyexi, could you help @thramu on how to fix this problem? Is this a valid linter issue?

@jianyexi  Can you please help me on the fix?

**Comment by tadelesh** [comment]
Since the change in this PR is not included in the default tag, approved the breaking changes for Go which are not related with this PR.

**Review by lirenhe — APPROVED** [review]

**Pull request merged**
